diff --git a/modules/jsonapi_defaults/src/Controller/EntityResource.php b/modules/jsonapi_defaults/src/Controller/EntityResource.php
index a268ecf62fbca321ffb7a5af18efa91a8761802d..f026b0e5cd358e5e79800f2b3b7f56561fc297e4 100644
--- a/modules/jsonapi_defaults/src/Controller/EntityResource.php
+++ b/modules/jsonapi_defaults/src/Controller/EntityResource.php
@@ -20,6 +20,7 @@ use Drupal\jsonapi\ResourceType\ResourceType;
 use Drupal\jsonapi\ResourceType\ResourceTypeRepositoryInterface;
 use Drupal\jsonapi_defaults\JsonapiDefaultsInterface;
 use Drupal\jsonapi_extras\Entity\JsonapiResourceConfig;
+use Symfony\Component\EventDispatcher\EventDispatcherInterface;
 use Symfony\Component\HttpFoundation\Request;
 use Symfony\Component\Serializer\SerializerInterface;
 
@@ -67,6 +68,8 @@ class EntityResource extends JsonApiEntityResource {
    *   The time service.
    * @param \Drupal\Core\Session\AccountInterface $user
    *   The current user account.
+   * @param \Symfony\Component\EventDispatcher\EventDispatcherInterface $event_dispatcher
+   *    The event dispatcher.
    * @param \Drupal\jsonapi_defaults\JsonapiDefaultsInterface $jsonapi_defaults
    *   The jsonapi default service.
    * @param \Drupal\Core\Logger\LoggerChannelFactoryInterface $loggerFactory
@@ -84,6 +87,7 @@ class EntityResource extends JsonApiEntityResource {
     SerializerInterface $serializer,
     TimeInterface $time,
     AccountInterface $user,
+    EventDispatcherInterface $event_dispatcher,
     JsonapiDefaultsInterface $jsonapi_defaults,
     LoggerChannelFactoryInterface $loggerFactory,
   ) {
@@ -99,6 +103,7 @@ class EntityResource extends JsonApiEntityResource {
       $serializer,
       $time,
       $user,
+      $event_dispatcher,
     );
     $this->jsonapiDefaults = $jsonapi_defaults;
     $this->logger = $loggerFactory->get('jsonapi_defaults');
